tower-http/src/lib.rs

```diff
@@ -37,3 +37,51 @@
 #![forbid(unsafe_code)]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![cfg_attr(test, allow(clippy::float_cmp))]
+
+use http::Response;
+
+#[cfg(feature = "trace")]
+#[cfg_attr(docsrs, doc(cfg(feature = "trace")))]
+pub mod trace;
+
+#[derive(Debug, Clone, Copy)]
+pub enum LatencyUnit {
+    Millis,
+    Nanos,
+}
+
+/// You might not always want to log the literal HTTP status. gRPC for example has its own status
+/// and always uses `200 OK` even for errors.
+// TODO(david): can we come up with a better name for this?
+pub trait GetTraceStatus<T, E> {
+    fn trace_status(&self, result: &Result<T, E>) -> TraceStatus;
+}
```

Comment:
This seems like it might be worth generalizing into a more generic notion of ["response classification"], which might be useful for other purposes as well (e.g. retries). In Linkerd, we have [our own version of this](https://github.com/linkerd/linkerd2-proxy/blob/e0271b4ee4841ef86fcbdb34f86b0fe725c502e2/linkerd/http-classify/src/lib.rs), it might be worthwhile to try to design a general-purpose response classification trait for `tower-http`?